lib/upnp/upnp.go

```diff
@@ -178,15 +172,29 @@ USER-AGENT: syncthing/1.0
 
 	l.Debugln("Listening for UPnP response for device type", deviceType, "on", intf.Name)
 
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
 	// Listen for responses until a timeout is reached
+loop:
 	for {
 		resp := make([]byte, 65536)
+		if err := socket.SetDeadline(time.Now().Add(time.Second)); err != nil {
```

Comment:
100ms maybe? 1s is still annoying.

I assume there is no api with a context?

Reply:
Yeah not for UDP sockets I'm afraid. :( 250ms?